core/src/main/java/tachyon/client/TachyonFS.java

```diff
@@ -580,6 +586,28 @@ public synchronized ClientFileInfo getFileStatus(int fileId, boolean useCachedMe
     return getFileStatus(fileId, TachyonURI.EMPTY_URI, useCachedMetadata);
   }
 
+  private synchronized <K> ClientFileInfo getFileStatus(Map<K, ClientFileInfo> cache, K key,
+      int fileId, String path, boolean useCachedMetaData) throws IOException {
+    ClientFileInfo info = cache.get(key);
+    if (useCachedMetaData && info != null) {
+      return info;
+    }
+
+    info = mMasterClient.getFileStatus(fileId, path);
+    fileId = info.getId();
+    if (fileId == -1) {
+      cache.remove(key);
+      return null;
+    }
+    path = info.getPath();
+
+    // TODO: LRU
+    mIdToClientFileInfo.put(fileId, info);
```

Comment:
This function abstracts the cache get or else invalidate logic, but here it is hardcoded. Perhaps the updates can be done outside this function?